pkg/etcd/etcd.go

```diff
@@ -458,16 +459,21 @@ func (c *CDCEtcdClientImpl) CreateChangefeedInfo(ctx context.Context,
 	}
 	opsThen := []clientv3.Op{
 		clientv3.OpPut(infoKey, value),
-		clientv3.OpPut(upstreamEtcdKeyStr, string(upstreamData)),
 	}
 	if len(upstreamResp.Kvs) == 0 {
 		cmps = append(cmps,
 			clientv3.Compare(clientv3.ModRevision(upstreamEtcdKeyStr),
 				"=", 0))
+		opsThen = append(opsThen,
+			clientv3.OpPut(upstreamEtcdKeyStr, string(upstreamData)))
 	} else {
 		cmps = append(cmps,
-			clientv3.Compare(clientv3.ModRevision(upstreamInfoKey.String()),
+			clientv3.Compare(clientv3.ModRevision(upstreamEtcdKeyStr),
 				"=", upstreamResp.Kvs[0].ModRevision))
+		if !bytes.Equal(upstreamResp.Kvs[0].Value, upstreamData) {
+			opsThen = append(opsThen,
+				clientv3.OpPut(upstreamEtcdKeyStr, string(upstreamData)))
+		}
 	}
 
 	resp, err := c.Client.Txn(ctx, cmps, opsThen, TxnEmptyOpsElse)
```

Comment:
if save changefeed or upstream info to etcd failed, the error message should not be "changefeed already exists"， I was confused when I see this error message.
 Something like "save changefeed to meta store failed" would be better.